File: c/function_fail_2.c
// @expect error

#include "smack.h" //<assert.h>

int cap(int x) {
    int y = x;
    if (10 < x) {
        y = 10;
    }
    return y;
}

int main() {
    assert(cap(2) == 2);
    assert(cap(15) != 10);
    int x = __VERIFIER_nondet_int(); 
    assert(cap(x) <= 10);
    return 0;
}

File: rust/function_fail_2.rs
// @expect error

fn cap(x: usize) -> usize {
    let mut y = x;
    if 10 < x {
        y = 10;
    }
    return y;
}

fn main() {
    let two = cap(2);
    let ten = cap(15);
    assert!(two == 2);
    assert!(ten != 10);
    let x = __VERIFIER_nondet_int();
    assert!(x <= 10);
}
